tests/phpunit/tests/post/wpPostType.php

```diff
@@ -229,4 +229,47 @@ public function test_applies_registration_args_filters() {
 
 		$this->assertSame( 3, $action->get_call_count() );
 	}
+
+	/**
+	 * @ticket 56922
+	 *
+	 * @dataProvider data_should_have_custom_controller_properties
+	 *
+	 * @covers WP_Post_Type::set_props
+	 *
+	 * @param string $property_name   Property name.
+	 * @param string $controller_name Controller class name.
+	 */
+	public function test_should_allow_to_set_custom_revisions_and_autosaves_controllers_properties( $property_name, $controller_name ) {
+		$post_type = new WP_Post_Type( 'wp_template' );
+		$post_type->set_props(
+			array(
+				$property_name => $controller_name,
+			)
+		);
```

Comment:
Instead of changing existing post types, create a new one. Then the data passed in the provider can just be args passed to when creating a new post type.